src/Tools/ExternalAccess/RazorTest/RazorAnalyzerAssemblyResolverTests.cs

```diff
@@ -0,0 +1,153 @@
+using System.Runtime.CompilerServices;
+using System.Runtime.Loader;
+using System.Text;
+using System.Threading.Tasks;
+using Basic.Reference.Assemblies;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.Razor.UnitTests;
+
+public sealed class RazorAnalyzerAssemblyResolverTests : IDisposable
+{
+    private TempRoot TempRoot { get; } = new TempRoot();
+    private int InitialAssemblyCount { get; }
+
+    public RazorAnalyzerAssemblyResolverTests()
+    {
+        InitialAssemblyCount = AssemblyLoadContext.GetLoadContext(this.GetType().Assembly)!.Assemblies.Count();
+    }
+
+    public void Dispose()
+    {
+        TempRoot.Dispose();
+
+        // This test should not change the set of assemblies loaded in the current context.
+        var count = AssemblyLoadContext.GetLoadContext(this.GetType().Assembly)!.Assemblies.Count();
+        Assert.Equal(InitialAssemblyCount, count);
+    }
+
+    internal void CreateRazorAssemblies(string directory, string versionNumber = "1.0.0.0")
+    {
+        foreach (var simpleName in RazorAnalyzerAssemblyResolver.RazorAssemblyNames)
+        {
+            BuildOne(simpleName);
+        }
+
+        void BuildOne(string simpleName)
+        {
+            var i = simpleName.LastIndexOf('.');
+            var typeName = simpleName[(i + 1)..];
+            var source = $$"""
+                using System.Reflection;
+
+                [assembly: AssemblyVersion("{{versionNumber}}")]
+                [assembly: AssemblyFileVersion("{{versionNumber}}")]
+
+                public sealed class {{typeName}} { }
+                """;
+
+            var compilation = CSharpCompilation.Create(
+                simpleName,
+                [CSharpSyntaxTree.ParseText(source)],
+                NetStandard20.References.All,
+                new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary));
+            var result = compilation.Emit(Path.Combine(directory, $"{simpleName}.dll"));
+            Assert.True(result.Success);
+        }
+    }
+
+    private static void RunWithLoader(Action<RazorAnalyzerAssemblyResolver, AnalyzerAssemblyLoader, AssemblyLoadContext> action)
+    {
+        var compilerLoadContext = new AssemblyLoadContext("Compiler", isCollectible: true);
+        var currentLoadContext = new AssemblyLoadContext("Current", isCollectible: true);
+        var loader = new AnalyzerAssemblyLoader([], [AnalyzerAssemblyLoader.DiskAnalyzerAssemblyResolver], compilerLoadContext);
+#pragma warning disable 612 
+        var resolver = CreateResolver();
+#pragma warning restore 612 
+        action(resolver, loader, currentLoadContext);
+
+        Assert.Empty(currentLoadContext.Assemblies);
+        currentLoadContext.Unload();
+        compilerLoadContext.Unload();
+    }
+
+    [Obsolete]
+    internal static RazorAnalyzerAssemblyResolver CreateResolver() => new RazorAnalyzerAssemblyResolver();
```

Comment:
Consider inlining this method at its only call site.

Reply:
Can't inline this because the ctor is obsolete with error. The only way to suppress that is to call it from an obsolete method.